text/0000-named-tracers.md

```diff
@@ -0,0 +1,91 @@
+# Named Tracers
+
+**Status:** `proposed`
+
+_Creating Tracers using a factory mechanism and naming those Tracers in accordance with the library that provides the instrumentation for a traced component._
+
+## Suggested reading
+
+* [Proposal: Tracer Components](https://github.com/open-telemetry/opentelemetry-specification/issues/10)
+* [Global Instance discussions](https://github.com/open-telemetry/opentelemetry-specification/labels/global%20instance)
+* [Proposal: Add a version resource](https://github.com/open-telemetry/oteps/pull/38)
+
+## Motivation
+
+The mechanism of "Named Tracers" proposed here is motivated by following scenarios:
+
+* For a consumer of OpenTelemetry instrumentation libraries, there is currently no possibility of influencing the amount of the data produced by such libraries. Instrumentation libraries can easily "spam" backend systems, deliver bogus data or - in the worst case - crash or slow down applications. These problems might even occur suddenly in production environments caused by external factors such as increasing load or unexpected input data.
+
+* If a library hasn't implemented [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md) correctly or those conventions change over time, it's currently hard to interpret and sanitize these data selectively. The produced Spans cannot associated with those instrumentation libraries later.
+
+This proposal attempts to solve the stated problems by introducing the concept of:
+ * _Named Tracers_ identified via a Resource (e.g. _"io.opentelemetry.contrib.mongodb"_, _"1.0.0"_) which is associated with the Tracer and the Spans it produces.
+ * A `TracerFactory` as the only means of creating a Tracer.
+
+Based on such an identifier, a Sampler could be implemented that discards Spans from certain libraries. Also, by providing a custom Exporter, Span data could be sanitized before it gets processed in a back-end system. However, this is beyond the scope of this proposal, which only provides the fundamental mechanisms.
+
+## Explanation
+
+From a user perspective, working with *Named Tracers* and *TracerFactories* is conceptually similar to how e.g. the [Java logging API](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#getLogger(java.lang.String)) and logging frameworks like [log4j](https://www.slf4j.org/apidocs/org/slf4j/LoggerFactory.html) work. In analogy to requesting Logger objects through LoggerFactories, a tracing library would create specific 'Tracer' objects through a 'TracerFactory'.
+
+Instead of using plain strings as an argument for creating new Tracers a Resource identifying an instrumentation library is used. Such resources must have a _version_ and _name_ labels (there could be semantic convention definitions for those labels). Version values will follow the conventions proposed [here](https://github.com/open-telemetry/oteps/pull/38).
```

Comment:
I am not sure I understand why a `Resource` is better here. My vote would be to use two strings, for the following reasons:
1. Simpler API to get a Tracer/Meter.
1. Limits what can be set and makes required fields more obviously.
1. Simpler API surface for OpenTelemetry - Resource does not need to be something that an instrumentation/library developer needs to learn/understand.
1. Component name can be used by some implementation as a prefix for Span name or Metric name. These implementations may not support a notion of Resource. This is specifically important for systems that do not support sub-target resources (components) and only target resources.

Reply:
Honestly, I don't have a strong opinion on this topic. Two strings would work equally well in my opinion.
To me this should come down to the decision of whether `Resource` should be part of the API or not:
* If Tracer creation is the only occurrence of this type in the API I vote in favor of using plain strings instead.
* If however Resources are a common API type, I see no reason why those should not also be used in this context.

Reply:
I would vote to not use Resource here to not bring back the Resource just for this, and use a "required" field name and an "optional" version (if version is useful).

Reply:
The main proposal uses plain strings again and resources as an alternative are mentioned in the "Prior art and alternatives" section (also referring to https://github.com/open-telemetry/opentelemetry-specification/pull/254)